src/main/java/com/nerdtranslator/translateapibridge/service/impl/TextToSpeechApiServiceImpl.java

```diff
@@ -1,52 +1,105 @@
-                                     .build())) {
-            SynthesisInput input = SynthesisInput.newBuilder()
-                    .setText(textToTransfer)
-                    .build();
-
-            VoiceSelectionParams voice = VoiceSelectionParams.newBuilder()
-                    .setLanguageCode(langCode)
-                    .setSsmlGender(SsmlVoiceGender.NEUTRAL)
-                    .build();
-
-            AudioConfig audioConfig = AudioConfig.newBuilder()
-                    .setAudioEncoding(AudioEncoding.MP3)
-                    .build();
-
-            SynthesizeSpeechResponse response = textToSpeechClient.synthesizeSpeech(input, voice, audioConfig);
-            ByteString audioRepresentationOfText = response.getAudioContent();
-            speechResult = audioRepresentationOfText.toByteArray();
+        byte[] speechResult = null;
+        TextToSpeechClient textToSpeechClient = null;
+        AudioEncoding audioEncoding = null;
+        try {
+            audioEncoding = AudioEncoding.MP3;
+            textToSpeechClient = configureTextToSpeechClient();
+            speechResult = getSpeechResult(textToSpeechClient, SsmlVoiceGender.NEUTRAL,
+                    audioEncoding, textToTransfer, langCode);
+        } catch (InvalidArgumentException e) {
+            byte[] result = isSuccessfulSpeechResultAgain(textToSpeechClient, audioEncoding, textToTransfer, langCode);
+            if (result != null) {
+                speechResult = result;
+            } else {
+                throw new VoiceGenderNotSupportedException("\"Voice isn't supported yet\"");
+            }
         } catch (IOException e) {
-            throw new RuntimeException("An error with text to speech service");
+            throw new RuntimeException("\"The result of text to speech translation wasn't received\"");
+        } finally {
+            if (textToSpeechClient != null) {
+                textToSpeechClient.close();
+            }
         }
         if (speechResult == null) {
-            throw new RuntimeException("The result of text to speech translation wasn't received");
+            throw new RuntimeException("\"The result of text to speech translation wasn't received\"");
         }
+
         return speechResult;
     }
+
+    private byte[] isSuccessfulSpeechResultAgain(TextToSpeechClient textToSpeechClient, AudioEncoding audioEncoding,
+                                                 String textToTransfer, String langCode) {
+        byte[] result;
+        List<SsmlVoiceGender> genders = new ArrayList<>(List.of(SsmlVoiceGender.FEMALE, SsmlVoiceGender.MALE));
+        for (SsmlVoiceGender gender : genders) {
+            try {
+                result = getSpeechResult(textToSpeechClient, gender, audioEncoding, textToTransfer, langCode);
+                return result;
+            } catch (InvalidArgumentException e) {
+                continue;
```

Comment:
throw new VoiceGenderNotSupportedException("\"Voice isn't supported yet\""); and add exception handling method in GlobalExceptionHandler with just logging, ot just log it here for now